Amin masharqa

AminMasharqa opened the pull request:
### Pull Request Title: Add Ynet News Scraper

### Description

This pull request introduces a new script to scrape news headlines and their publication times from the Ynet news website. The script fetches data from the specified URL, parses the HTML to extract relevant information, and saves it to a JSON file. 

### Changes Made:
- **Added `scrape_ynet` function**:
  - Fetches HTML content from `https://www.ynetnews.com/category/3089`.
  - Uses `BeautifulSoup` to parse the HTML content.
  - Extracts publication dates and titles of news items.
  - Converts date strings to `datetime` objects and formats them.
  - Stores news items in a dictionary with the publication datetime as keys and titles as values.
  - Sorts the dictionary by datetime in descending order.
  - Saves the sorted news items to a JSON file (`2024-07-28.json`).

### Files Added/Modified:
- **`scrape_ynet.py`**: Contains the implementation of the Ynet news scraper.

### Requirements:
- **Dependencies**:
  - `requests`
  - `beautifulsoup4`
- **Standard Libraries**:
  - `datetime`
  - `json`

### Usage:
1. Ensure all dependencies are installed:
   ```bash
   pip install requests beautifulsoup4
   ```
2. Run the script:
   ```bash
   python scrape_ynet.py
   ```

### Additional Notes:
- The script currently saves the output to a JSON file named with the current date (`2024-07-28.json`). This can be modified as needed.
- The URL and HTML structure used are specific to the Ynet news site and may need adjustments if the site structure changes.

### Testing:
- The script was tested by running it locally and verifying that the JSON file is generated correctly with the expected data format.

### Request for Review:
- Please review the code for any potential improvements or optimizations.
- Verify that the script meets the project requirements and coding standards.

Thank you!

---

ShakedZrihen commented on assignments/first-class-assignment-scrapers/AminMasharqa/2024-07-28.json:
```diff
@@ -0,0 +1,52 @@
+{
+    "2024-07-28 20:13:33": "12th victim of Hezbollah rocket attack on Majdal Shams confirmed",
```
please see requested json format:

filename: date
content:

```
{
    "hh:mm": content
}
```

ShakedZrihen commented on assignments/first-class-assignment-scrapers/AminMasharqa/scraper.py:
```diff
@@ -0,0 +1,51 @@
+import json
+from datetime import datetime
+
+def scrape_ynet():
```
this function is too long,
im suggesting to split it to:

get_content_from_remote:
fetch html 
return soup handler


parse_one_item:
the body of the for

save_to_file:
the lines that saves data into file


than the scrape function can call these methods :)

AminMasharqa commented:
Hi Shaked,

I have made the requested changes to the `scrape_ynet` function to align with your feedback. Specifically, I have refactored the code into smaller, more manageable functions and updated the JSON format. The modifications include:

1. **Function `get_content_from_remote`**:
    - Fetches the HTML content and returns a BeautifulSoup object.

2. **Function `parse_one_item`**:
    - Parses a single news item and returns the time and title.

3. **Function `save_to_file`**:
    - Saves the dictionary to a JSON file in the specified format.

Additionally, I have adjusted the JSON structure to match the required format with `hh:mm` as keys.

The updated script has been tested and works correctly. Please review the changes at your convenience.

Thank you!

Best regards,
Amin Masharqa
